### PR TITLE
cli: Add jenkinsctl to repo

### PR DESCRIPTION
**Pull request by dougsland**
This PR adds to the project a CLI based on gojenkins library.
The initial code provides cover for the most gojenkins code.

Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>

### CODE REVIEW

**Comment by dougsland** [comment]
As no reviewers added to the PR, adding @figo to the loop.

File: cli/jenkinsctl/Makefile [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+createmod: clean
+	cd jenkins && go mod init
+	cd jenkins && go mod tidy
+	go mod init
+	go mod tidy
+
+all: createmod build
+
+clean:
+	rm -f go.sum go.mod $(BIN_ALL_PLATFORMS)
```
Comment (figo) [inline cli/jenkinsctl/Makefile:38]: I am curious why want to remove go.mod file during cleanup, it is possible to have redirect or other special setting in go.mod

**Review by figo — COMMENTED** [review]
Looks like cli is a good way to interact with Jenkins server, thank you for the effort.

File: cli/jenkinsctl/Makefile (hunk 1 above)
Comment (dougsland) [reply cli/jenkinsctl/Makefile:38]: it was part of my development effort to make sure I was using the latest bits. I will remove it. Any other comment? @figo

**Comment by figo** [comment]
looks good to me overall, will go through 2nd round shortly

**Pull request merged**